source/iosuhax_devoptab.c

```diff
@@ -762,17 +877,21 @@ static int fs_dev_dirnext_r(struct _reent *r, DIR_ITER *dirState, char *filename
 
     if (st) {
         memset(st, 0, sizeof(struct stat));
-        st->st_mode   = (dir_entry->stat.flag & 0x80000000) ? S_IFDIR : S_IFREG;
-        st->st_nlink  = 1;
-        st->st_size   = dir_entry->stat.size;
-        st->st_blocks = (dir_entry->stat.size + 511) >> 9;
-        st->st_dev    = dir_entry->stat.id;
-        st->st_uid    = dir_entry->stat.owner_id;
-        st->st_gid    = dir_entry->stat.group_id;
-        st->st_ino    = dir_entry->stat.id;
-        st->st_atime  = dir_entry->stat.mtime;
-        st->st_ctime  = dir_entry->stat.ctime;
-        st->st_mtime  = dir_entry->stat.mtime;
+
+        // Convert fields to posix stat
+        st->st_dev     = (dev_t) dirIter->dev;
+        st->st_ino     = dir_entry->info.entryId;
+        st->st_mode    = fs_dev_translate_mode(dir_entry->info, true, false);
+        st->st_nlink   = 1;
+        st->st_uid     = dir_entry->info.owner;
+        st->st_gid     = dir_entry->info.group;
+        st->st_rdev    = st->st_dev;
+        st->st_size    = dir_entry->info.size;
+        st->st_blksize = 512;
+        st->st_blocks  = (st->st_size + st->st_blksize - 1) >> 9;
```

Comment:
Where is that 9 coming? This looks like a hardcoded blocksize of 512? 
Is the blocksize **always** 512?

doing `st->st_blocks  = (st->st_size + st->st_blksize - 1) / st->st_blksize;` feels a bit better (even though the st_blksize is hardcoded anyway).